mp/src/materialsystem/stdshaders/pbr_dx9.cpp

```diff
@@ -271,24 +290,52 @@ BEGIN_VS_SHADER(PBR, "PBR shader")
 			else
 			{
 				// We need the position, surface normal, and vertex compression format
-                unsigned int flags = VERTEX_POSITION | VERTEX_NORMAL;
+				unsigned int flags = VERTEX_POSITION | VERTEX_NORMAL | VERTEX_TANGENT_S | VERTEX_TANGENT_T;
 				// We only need one texcoord, in the default float2 size
 				pShaderShadow->VertexShaderVertexFormat(flags, 3, 0, 0);
 			}
+#ifdef PBRPARALLAX
+			const int useParallax = params[info.useParallax]->GetIntValue();
+#endif
+			if (!g_pHardwareConfig->SupportsShaderModel_3_0() || mat_pbr_force_20b.GetBool())
```

Comment:
Why didn't you use the fallback system?

Reply:
i read VertexLitGeneric shader and this is how valve does it.